src/Symfony/Component/Debug/ErrorHandler.php

```diff
@@ -215,9 +225,23 @@ public function setLoggers(array $loggers)
                 throw new \InvalidArgumentException('Invalid logger provided');
             }
             $this->loggers[$type] = $log + $prev[$type];
+
+            if ($this->bootstrappingLogger && $prev[$type][0] === $this->bootstrappingLogger) {
+                $flush[$type] = $type;
+            }
         }
         $this->reRegister($prevLogged | $this->thrownErrors);
 
+        if ($flush) {
+            foreach ($this->bootstrappingLogger->cleanLogs() as $log) {
+                if (!isset($flush[$type = $log[2]['type']])) {
```

Comment:
I suggest moving the assignment outside of the array access to make the code more readable. The current expression makes it kinda hard to read.